ibm/resource_ibm_firewall_shared.go

```diff
@@ -0,0 +1,224 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const (
+	FwHardwarePackageType = "ADDITIONAL_SERVICES_FIREWALL"
+)
+
+func resourceIBMFirewallShared() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMFirewallSharedCreate,
+		Read:     resourceIBMFirewallSharedRead,
+		Delete:   resourceIBMFirewallSharedDelete,
+		Exists:   resourceIBMFirewallSharedExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"billing_item_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"firewall_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
Add validateFunc to validate the input 
	ValidateFunc: validateAllowedStringValue([]string{"10MBPS_HARDWARE_FIREWALL", "20MBPS_HARDWARE_FIREWALL", "100MBPS_HARDWARE_FIREWALL", "1024MBPS_HARDWARE_FIREWALL"}),